### ISSUE TITLE
createConnection vs. connect

### ISSUE THREAD

**Issue by akoskm** [jedireza/drywall#352]
Hello!

I'm wondering what's the benefit of using `mongoose.createConnection` here: https://github.com/jedireza/drywall/blob/master/app.js#L27. If I'm no mistaken, with `mongo.connect` we could use mongoose to register our models like this:

```js
var mongoose = require('mongoose');
exports = module.exports = function(app) {

  'use strict';
  var userSchema = new mongoose.Schema({
  });
  mongoose.model('User', userSchema);
};
```

and there would be no need to carry `app.db` around and inject it into every single module where we need a model from mongoose, like in: https://github.com/jedireza/drywall/blob/master/models.js.

**Comment by jedireza** [comment]
Thanks for opening an issue. This sounds like a good plan. Would you like to submit a PR for this?

**Comment by akoskm** [comment]
I'm using the [angular-drywall](https://github.com/arthurkao/angular-drywall) fork of this project, but I can do the refactoring here too.

How we could test the refactored code? In my projects (mostly drywall-based stuff) I'm using https://github.com/visionmedia/supertest but I also have some default data to work with. This makes integration testing easier since I can do a full _login-dosomething-logout_ cycle.

I think we could use mockgoose to create such default data before initializing the app and running the tests. Currently it has some limitations with db connections created with `createConnection` (see https://github.com/mccormicka/Mockgoose/issues/195), but that shouldn't cause any problem after this refactoring.

What do you think?

**Comment by jedireza** [comment]
> How we could test the refactored code?

Good question. When I created Drywall I never had tests and because of this it needs to be de-coupled a bit from itself. I wised up since then and for new projects like [Frame](https://github.com/jedireza/frame/), tests + coverage are top priorities.

Refactoring everything at once is a big task and unlikely to get done soon. Maybe this is a good chance to make the changes and create some basic tests for the schemas.
